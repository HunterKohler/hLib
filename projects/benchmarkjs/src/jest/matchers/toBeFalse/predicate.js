export default recieved => recieved === false;