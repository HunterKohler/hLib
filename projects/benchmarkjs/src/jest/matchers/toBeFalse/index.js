import { matcherHint, printReceived, printExpected } from "jest-matcher-utils";
import predicate from "./predicate";

const passMessage = recieved => () =>
    matcherHint(".not.toBeFalse", "recieved", "") 
    + "\n\n"
    + "Expected value to not be false, recieved:\n"
    + `    ${printReceived(recieved)}`;

const failMessage = recieved => () =>
    matcherHint(".toBeFalse", "recieved", "")
    + "\n\n"
    + "Expected value to be false:\n"
    + `    ${printExpected(false)}\n`
    + "Recieved:\n"
    + `    ${printReceived(recieved)}`;

export default function toBeFalse(recieved) {
    const pass = predicate(recieved);
    return { pass, message: pass
        ? passMessage(recieved)
        : failMessage(recieved)
    };
}