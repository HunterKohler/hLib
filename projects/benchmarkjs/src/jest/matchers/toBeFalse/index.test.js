import toBeFalse from "./";

expect.extend({ toBeFalse });

describe(".toBeFalse", () => {
    test("pass when given false", () => {
        expect(false).toBeFalse();
    });

    test.each([
        [true], [""], [0], [1], [{}], [[]], [undefined], [null], [NaN]
    ])("fail when given %s", value => {
        expect(() => expect(value).toBeFalse()).toThrowErrorMatchingSnapshot();
    });
})

describe(".not.toBeFalse", () => {
    test("fail when given false", () => {
        expect(() => expect(false).not.toBeFalse()).toThrowErrorMatchingSnapshot();
    });

    test.each([
        [true], [""], [0], [1], [{}], [[]], [undefined], [null], [NaN]
    ])("pass when given %s", value => {
        expect(value).not.toBeFalse();
    });
});