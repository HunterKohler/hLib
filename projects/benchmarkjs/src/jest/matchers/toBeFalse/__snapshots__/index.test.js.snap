// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeFalse fail when given false 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).not.toBeFalse()[22m

Expected value to not be false, recieved:
    [31mfalse[39m"
`;

exports[`.toBeFalse fail when given  1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31m\\"\\"[39m"
`;

exports[`.toBeFalse fail when given [] 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31m[][39m"
`;

exports[`.toBeFalse fail when given {} 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31m{}[39m"
`;

exports[`.toBeFalse fail when given 0 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31m0[39m"
`;

exports[`.toBeFalse fail when given 1 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31m1[39m"
`;

exports[`.toBeFalse fail when given NaN 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31mNaN[39m"
`;

exports[`.toBeFalse fail when given null 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31mnull[39m"
`;

exports[`.toBeFalse fail when given true 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31mtrue[39m"
`;

exports[`.toBeFalse fail when given undefined 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeFalse()[22m

Expected value to be false:
    [32mfalse[39m
Recieved:
    [31mundefined[39m"
`;
